kolibri/plugins/management/assets/src/actions.js

```diff
@@ -320,19 +328,31 @@ function cancelImportExportWizard(store) {
 
 // called from a timer to continually update UI
 function pollTasksAndChannels(store) {
-  TaskResource.getCollection().fetch({}, true).then(
+  const samePageCheck = samePageCheckGenerator(store);
+  TaskResource.getCollection().fetch({}, true).only(
+    () => {
+      // don't handle response if we've switched pages
+      if (!samePageCheck()) {
+        return false;
+      }
+      // don't handle response if we're in the middle of another operation
+      return !store.state.pageState.wizardState.busy;
+    },
     (taskList) => {
-      ChannelResource.getCollection({}).fetch({}, true).then((channelList) => {
-        store.dispatch('SET_CONTENT_PAGE_TASKS', taskList.map(_taskState));
-        store.dispatch('SET_CONTENT_PAGE_CHANNELS', channelList);
-
-        // Close the wizard if there's an outstanding task.
-        // (this can be removed when we support more than one
-        // concurrent task.)
-        if (taskList.length && store.state.pageState.wizardState.shown) {
-          cancelImportExportWizard(store);
+      ChannelResource.getCollection({}).fetch({}, true).only(
```

Comment:
They're done serially to prevent a chance of a race condition where the list of channels might update _before_ the task display updates. I'll add a comment explaining this.